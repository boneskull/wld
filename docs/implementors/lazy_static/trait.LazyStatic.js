(function() {var implementors = {};
implementors["wld"] = [{"text":"impl LazyStatic for BLOCKTYPE_COLOR_MAP","synthetic":false,"types":[]},{"text":"impl LazyStatic for WALLTYPE_COLOR_MAP","synthetic":false,"types":[]},{"text":"impl LazyStatic for PAINT_COLOR_MAP","synthetic":false,"types":[]}];
if (window.register_implementors) {window.register_implementors(implementors);} else {window.pending_implementors = implementors;}})()