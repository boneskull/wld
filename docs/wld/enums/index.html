<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="API documentation for the Rust `enums` mod in crate `wld`."><meta name="keywords" content="rust, rustlang, rust-lang, enums"><title>wld::enums - Rust</title><link rel="stylesheet" type="text/css" href="../../normalize.css"><link rel="stylesheet" type="text/css" href="../../rustdoc.css" id="mainThemeStyle"><link rel="stylesheet" type="text/css" href="../../light.css"  id="themeStyle"><link rel="stylesheet" type="text/css" href="../../dark.css" disabled ><link rel="stylesheet" type="text/css" href="../../ayu.css" disabled ><script src="../../storage.js"></script><noscript><link rel="stylesheet" href="../../noscript.css"></noscript><link rel="icon" type="image/svg+xml" href="../../favicon.svg">
<link rel="alternate icon" type="image/png" href="../../favicon-16x16.png">
<link rel="alternate icon" type="image/png" href="../../favicon-32x32.png"><style type="text/css">#crate-search{background-image:url("../../down-arrow.svg");}</style></head><body class="rustdoc mod"><!--[if lte IE 8]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="sidebar"><div class="sidebar-menu">&#9776;</div><a href='../../wld/index.html'><div class='logo-container rust-logo'><img src='../../rust-logo.png' alt='logo'></div></a><p class="location">Module enums</p><div class="sidebar-elems"><div class="block items"><ul><li><a href="#enums">Enums</a></li></ul></div><p class="location"><a href="../index.html">wld</a></p><script>window.sidebarCurrent = {name: "enums", ty: "mod", relpath: "../"};</script><script defer src="../sidebar-items.js"></script></div></nav><div class="theme-picker"><button id="theme-picker" aria-label="Pick another theme!"><img src="../../brush.svg" width="18" alt="Pick another theme!"></button><div id="theme-choices"></div></div><script src="../../theme.js"></script><nav class="sub"><form class="search-form"><div class="search-container"><div><select id="crate-search"><option value="All crates">All crates</option></select><input class="search-input" name="search" disabled autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"></div><span class="help-button">?</span>
                <a id="settings-menu" href="../../settings.html"><img src="../../wheel.svg" width="18" alt="Change settings"></a></div></form></nav><section id="main" class="content"><h1 class="fqn"><span class="out-of-band"><span id="render-detail"><a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">[<span class="inner">&#x2212;</span>]</a></span><a class="srclink" href="../../src/wld/enums.rs.html#1-26" title="goto source code">[src]</a></span><span class="in-band">Module <a href="../index.html">wld</a>::<wbr><a class="mod" href="">enums</a></span></h1><h2 id="enums" class="section-header"><a href="#enums">Enums</a></h2>
<table><tr class="module-item"><td><a class="enum" href="enum.AnglerQuestFish.html" title="wld::enums::AnglerQuestFish enum">AnglerQuestFish</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.BlockShape.html" title="wld::enums::BlockShape enum">BlockShape</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.BlockType.html" title="wld::enums::BlockType enum">BlockType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.EntityType.html" title="wld::enums::EntityType enum">EntityType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.EvilType.html" title="wld::enums::EvilType enum">EvilType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.HardmodeOreType.html" title="wld::enums::HardmodeOreType enum">HardmodeOreType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.InvasionType.html" title="wld::enums::InvasionType enum">InvasionType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.ItemType.html" title="wld::enums::ItemType enum">ItemType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.LiquidType.html" title="wld::enums::LiquidType enum">LiquidType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.RLEType.html" title="wld::enums::RLEType enum">RLEType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.TBool.html" title="wld::enums::TBool enum">TBool</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.TileEntityType.html" title="wld::enums::TileEntityType enum">TileEntityType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.WallType.html" title="wld::enums::WallType enum">WallType</a></td><td class="docblock-short"><p>A list of vanilla Walls by internal name, mapped to their IDs.</p>
</td></tr></table></section><section id="search" class="content hidden"></section><section class="footer"></section><script>window.rootPath = "../../";window.currentCrate = "wld";</script><script src="../../main.js"></script><script defer src="../../search-index.js"></script></body></html>