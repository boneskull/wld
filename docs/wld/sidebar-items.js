initSidebarItems({"fn":[["enable_logger","Enable logging for this crate given [`log::LevelFilter`] `level`."],["parse_world","Instantiate a [`World`] from a slice of bytes."]],"mod":[["constants",""],["enums",""],["models",""]]});