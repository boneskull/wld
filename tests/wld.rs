use insta::assert_debug_snapshot;
use lazy_static::lazy_static;
use std::fs::read;
use wld::{model::world::World, parse_world};

lazy_static! {
  // paths are relative to root, I guess?
  pub static ref WORLD: std::vec::Vec<u8> =
    read("tests/fixtures/Foon.wld").expect("Unable to read file");
  pub static ref PARSED_WORLD: World = parse_world(&WORLD).unwrap();
}

#[test]
fn test_parse_header() {
  assert_debug_snapshot!(PARSED_WORLD.header);
}

#[test]
fn test_parse_properties() {
  assert_debug_snapshot!(PARSED_WORLD.properties);
}

#[test]
fn test_parse_status() {
  assert_debug_snapshot!(PARSED_WORLD.status);
}
