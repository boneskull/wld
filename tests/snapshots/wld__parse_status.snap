---
source: tests/wld.rs
expression: PARSED_WORLD.status
---
WorldStatus {
    header: Header {
        version: "1.3.5.3",
        revision: 160,
        is_favorite: false,
        offsets: Offsets {
            header: 127,
            tiles: 2802,
            chests: 2860224,
            signs: 2879758,
            npcs: 2880141,
            tile_entities: 2880453,
            pressure_plates: 2880457,
            town_manager: 2880461,
            footer: 2880489,
        },
    },
    properties: Properties {
        tile_frame_importances: VariableTBitVec(
            BitVec<Lsb0, u8> [
                0b00011100,
                0b00111111,
                0b11111100,
                0b10111101,
                0b01111000,
                0b00100000,
                0b00100001,
                0b00000100,
                0b00000001,
                0b11100111,
                0b01111111,
                0b11111111,
                0b11111111,
                0b11100010,
                0b01100000,
                0b00000110,
                0b11001111,
                0b11110111,
                0b10000100,
                0b00000000,
                0b00000100,
                0b00011110,
                0b00100000,
                0b11110000,
                0b00000000,
                0b01000001,
                0b01101001,
                0b11111000,
                0b00011001,
                0b01011111,
                0b11111111,
                0b00000010,
                0b00000000,
                0b00000111,
                0b00011111,
                0b11110111,
                0b11111111,
                0b11111111,
                0b11111110,
                0b00101111,
                0b10011000,
                0b00000011,
                0b01110000,
                0b00000100,
                0b00111011,
                0b11111000,
                0b00001111,
                0b11101000,
                0b00111111,
                0b11110000,
                0b00000110,
                0b00111110,
                0b00011001,
                0b11011100,
                0b00011111,
                0b11111100,
                0b00001111,
                0b11000111,
                0b11111100,
            ],
        ),
        name: TString(
            "Foon",
        ),
        generator: GeneratorInfo {
            seed: TString(
                "1451234789",
            ),
            version: 833223655425,
        },
        uuid: TUuid(
            d578e106-3827-f648-a224-254c06ca78cb,
        ),
        id: 1468463142,
        bounds: Rect {
            left: 0,
            right: 67200,
            top: 0,
            bottom: 19200,
        },
        height: 1200,
        width: 4200,
        is_expert: False,
        created_on: 9860045932737703464,
        style: WorldStyle {
            moon: 1,
            trees: QuadrantStyle {
                x1: 3072,
                x2: 4200,
                x3: 4200,
                far_left: 4,
                near_left: 5,
                near_right: 0,
                far_right: 0,
            },
            moss: QuadrantStyle {
                x1: 1210,
                x2: 4200,
                x3: 4200,
                far_left: 1,
                near_left: 0,
                near_right: 3,
                far_right: 0,
            },
            underground_snow: 3,
            underground_jungle: 0,
            hell: 0,
        },
        spawn_point: Point {
            x: 2098,
            y: 229,
        },
        underground_level: 300.0,
        cavern_level: 528.0,
        current_time: 34958.0,
        is_daytime: True,
        moon_phase: 1,
        is_blood_moon: False,
        is_eclipse: False,
        dungeon_point: Point {
            x: 3426,
            y: 211,
        },
        evil_type: Corruption,
    },
    status: Status {
        bosses_slain: BossesSlain {
            eye_of_cthulhu: True,
            eater_of_worlds: True,
            skeletron: False,
            queen_bee: True,
            the_twins: False,
            the_destroyer: False,
            skeletron_prime: False,
            mechanical_boss: False,
            plantera: False,
            golem: False,
            king_slime: True,
        },
        saved_npcs: SavedNPCs {
            goblin_tinkerer: True,
            wizard: False,
            mechanic: False,
        },
        events_completed: EventsCompleted {
            goblin_army: True,
            clown: False,
            frost_moon: False,
            pirates: False,
        },
        shadow_orb_status: ShadowOrbStatus {
            smashed: True,
            meteorite_spawned: False,
            evil_boss_counter: 0,
        },
        smashed_altar_count: 0,
        is_hardmode: False,
        invasion_status: InvasionStatus {
            delay: 0,
            size: 0,
            invasion_type: NoInvasion,
            position: 0.000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001371372158997,
            slime_rain_time_left: 0.0,
        },
        sundial_cooldown: 0,
        rain_status: RainStatus {
            is_active: False,
            time_left: 0,
            max_rain: NaN,
        },
        hardmode_ore_1: UnknownOre,
        hardmode_ore_2: UnknownOre,
        hardmode_ore_3: UnknownOre,
        backgrounds: Backgrounds {
            forest: 4,
            corruption: 1,
            jungle: 1,
            snow: 0,
            hallow: 2,
            crimson: 24,
            desert: 34,
            ocean: 1,
        },
        clouds: Clouds {
            background: -1577056768,
            count: -14943,
            wind_speed: 0.000000000000000000000000000000000000000000265,
        },
        angler_quest_status: AnglerQuestStatus {
            completed_players: [
                TString(
                    "",
                ),
            ],
            angler_saved: True,
            target: EaterOfPlankton,
        },
        stylist_saved: False,
        tax_collector_saved: False,
        invasion_size_start: 120,
        cultist_delay: 0,
        mob_kills: MobKills(
            [
                0,
                0,
                0,
                0,
                9,
                0,
                0,
                85,
                0,
                0,
                0,
                0,
                0,
                1,
                8,
                0,
                0,
                0,
                2,
                4,
                0,
                0,
                0,
                0,
                0,
                102,
                0,
                281,
                0,
                10,
                0,
                0,
                5,
                0,
                0,
                0,
                0,
                0,
                36,
                45,
                34,
                4,
                69,
                4,
                0,
                2,
                0,
                7,
                0,
                0,
                1,
                8,
                0,
                0,
                2,
                16,
                0,
                0,
                0,
                0,
                0,
                14,
                0,
                0,
                11,
                0,
                0,
                3,
                0,
                271,
                4,
                31,
                0,
                0,
                0,
                0,
                3,
                0,
                0,
                5,
                1,
                0,
                0,
                0,
                13,
                0,
                0,
                264,
                0,
                0,
                20,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                9,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                45,
                202,
                0,
                0,
                0,
                0,
                52,
                0,
                31,
                0,
                0,
                0,
                67,
                46,
                0,
                1,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                7,
                0,
                0,
                0,
                0,
                2,
                0,
                0,
                0,
                27,
                0,
                0,
                0,
                53,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                9,
                0,
                0,
                12,
                2,
                0,
                0,
                0,
                2,
                0,
                1,
                12,
                0,
                0,
                0,
                21,
                0,
                0,
                0,
                1,
                0,
                2,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                2,
                25,
                5,
                38,
                40,
                9,
                0,
                14,
                20,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                0,
                9,
                0,
                0,
                0,
                0,
                0,
                0,
                4,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
            ],
        ),
        fast_forward_time: False,
        bosses_slain_2: BossesSlain2 {
            duke_fishron: False,
            martian_madness: False,
            lunatic_cultist: False,
            moon_lord: False,
            pumpking: False,
            mourning_wood: False,
            ice_queen: False,
            santa_nk1: False,
            everscream: False,
            solar_pillar: False,
            vortex_pillar: False,
            nebula_pillar: False,
            stardust_pillar: False,
        },
        pillar_status: PillarStatus {
            solar: False,
            vortex: False,
            nebula: False,
            stardust: False,
            is_active: False,
        },
        party_status: PartyStatus {
            party_center_is_active: False,
            natural_party_is_active: False,
            party_cooldown: 3,
            partying_npcs: PartyingNPCs(
                [],
            ),
        },
        sandstorm_status: SandstormStatus {
            is_active: False,
            time_left: 0,
            severity: 0.19280446,
            intended_severity: 0.19280446,
        },
        bartender_saved: True,
        old_ones_army_status: OldOnesArmyStatus {
            tier1: False,
            tier2: False,
            tier3: False,
        },
    },
}
